safe_transaction_service/history/tasks.py

```diff
@@ -534,25 +533,42 @@ def retry_get_metadata_task(
 
     # Maybe other task already retrieved the metadata
     cached_metadata = collectible_with_metadata_cached["metadata"]
-    metadata = (
-        cached_metadata
-        if cached_metadata
-        else collectibles_service.get_metadata(collectible)
-    )
-    collectible_with_metadata = CollectibleWithMetadata(
-        collectible.token_name,
-        collectible.token_symbol,
-        collectible.logo_uri,
-        collectible.address,
-        collectible.id,
-        collectible.uri,
-        metadata,
-    )
-
-    redis.set(
-        redis_key,
-        json.dumps(dataclasses.asdict(collectible_with_metadata)),
-        collectibles_service.COLLECTIBLE_EXPIRATION,
-    )
-
+    try:
+        metadata = (
+            cached_metadata
+            if cached_metadata
+            else collectibles_service.get_metadata(collectible)
+        )
+        collectible_with_metadata = CollectibleWithMetadata(
+            collectible.token_name,
+            collectible.token_symbol,
+            collectible.logo_uri,
+            collectible.address,
+            collectible.id,
+            collectible.uri,
+            metadata,
+        )
+        redis.set(
+            redis_key,
+            json.dumps(dataclasses.asdict(collectible_with_metadata)),
+            collectibles_service.COLLECTIBLE_EXPIRATION,
+        )
+    except MetadataRetrievalExceptionTimeout:
+        # Random avoid to run all tasks at the same time
+        if (
+            retry_get_metadata_task.request.retries
+            < retry_get_metadata_task.max_retries
+        ):
+            retry_get_metadata_task.retry(
+                countdown=int(
+                    random.uniform(55, 65) * retry_get_metadata_task.request.retries
+                )
+            )
+        else:
+            logger.debug(
+                "Cannot get the metadata from %s after %i retries",
+                collectible.uri,
+                retry_get_metadata_task.request.retries,
+            )
```

Comment:
```suggestion
            logger.debug(
                "Cannot get the metadata from %s after %i retries",
                collectible.uri,
                retry_get_metadata_task.request.retries,
                exc_info=True
            )
```

Reply:
Changed the message to include that was a timeout exception